src/decomp_2d_init_fin.f90

```diff
@@ -27,8 +19,9 @@
   !     all internal data structures initialised properly
   !     library ready to use
   !======================================================================
-  module subroutine decomp_2d_init_ref(nx,ny,nz,p_row,p_col,periodic_bc,comm)
+  subroutine decomp_2d_init_ref(nx,ny,nz,p_row,p_col,periodic_bc,comm)
 
+    use mpi
```

Comment:
Now that this file is ``#included`` is this still necessary? Sometimes Cray IIRC won't compile if a module is ``used`` twice